src/components/AppState/AppState.tsx

```diff
@@ -0,0 +1,138 @@
+import React, {
+  createContext,
+  useContext,
+  useState,
+  ReactNode,
+  useEffect,
+} from 'react';
+import { StatusFilter, Todo } from '../../types/Todo';
+import { handleErrorMessage } from '../function/handleErrorMessage ';
+import { getTodos } from '../../api/todos';
+
+export const USER_ID = 11956;
+
+export type AppStateContextType = {
+  todos: Todo[] | null;
+  setTodos: React.Dispatch<React.SetStateAction<Todo[] | null>>;
+  todosFilter: Todo[] | null;
+  setTodosFilter: React.Dispatch<React.SetStateAction<Todo[] | null>>;
+  tempTodo: Todo | null;
+  setTempTodo: React.Dispatch<React.SetStateAction<Todo | null>>;
+  errorNotification: string | null;
+  setErrorNotification: React.Dispatch<React.SetStateAction<string | null>>;
+  filter: StatusFilter;
+  setFilter: React.Dispatch<React.SetStateAction<StatusFilter>>;
+  loading: boolean;
+  setLoading: React.Dispatch<React.SetStateAction<boolean>>;
+};
+
+const AppStateContext = createContext<AppStateContextType | undefined>(
+  undefined,
+);
```

Comment:
null is better value for default state, cuz `undefined` - is like who knows what is that
```suggestion
const AppStateContext = createContext<AppStateContextType | null>(
  null,
);
```